gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -238,11 +241,37 @@ else if (statement instanceof ExecuteImmediate executeImmediate) {
         }
     }
 
-    private Map<String, String> getPreparedStatements(HttpServletRequest request)
+    private void processRequestBody(ContainerRequestContext requestContext)
+    {
+        InputStream entityStream = requestContext.getEntityStream();
+        if (entityStream == null ||
+                !requestContext.hasEntity() || requestContext.getMediaType() == null ||
+                !requestContext.getMediaType().getParameters().containsKey("charset") ||
+                !StandardCharsets.UTF_8.name().equalsIgnoreCase(requestContext.getMediaType().getParameters().get("charset"))) {
```

Comment:
How about separating to a different function and simplify it to

```suggestion
    if (!shouldProcessBody(requestContext)) {
        return;
    }

```
```java
private boolean shouldProcessBody(ContainerRequestContext requestContext) {
    if (!requestContext.hasEntity()) return false;
    MediaType mediaType = requestContext.getMediaType();
    if (mediaType == null) return false;
    String charset = mediaType.getParameters().get("charset");
    return StandardCharsets.UTF_8.name().equalsIgnoreCase(charset);
}
```